src/main/js/wiki/components/PageHeader.tsx

```diff
@@ -121,72 +111,72 @@ class PageHeader extends React.Component<Props, State> {
   };
 
   getPagePathWithoutRootDirectory(page, wiki) {
+    if (!page) {
+      return "";
+    }
+    if (page.path == undefined) {
+      return page;
+    }
     if (page.path.indexOf(wiki.directory) === 0) {
       return page.path.substring(wiki.directory.length + 1);
     }
-
     return page.path;
   }
-
   render() {
-    const { page, wiki, classes, onDelete, historyLink, branch, branches, pushBranchStateFunction } = this.props;
-
+    const {
+      page,
+      wiki,
+      classes,
+      onDelete,
+      historyLink,
+      inSettings,
+      inEdit,
+      pushBranchStateFunction,
+      branch,
+      branches
+    } = this.props;
     const pathWithoutRoot = this.getPagePathWithoutRootDirectory(page, wiki);
+    let editButton;
+    let historyButton;
+    let moveButton;
+    let deleteButton;
+    let restoreButton;
 
-    const editButton = page._links.edit ? (
-      <ActionLink glyphicon="glyphicon-edit" type="menu" to="?edit=true" i18nKey="page-header_edit" />
-    ) : (
-      ""
-    );
-    const createButton = (
-      <ActionButton glyphicon="glyphicon-plus" type="menu" onClick={this.onCreateClick} i18nKey="page-header_create" />
-    );
-    const historyButton = (
-      <ActionLink glyphicon="glyphicon-step-backward" type="menu" to={historyLink} i18nKey="page-header_history" />
-    );
-    const moveButton = page._links.move ? (
-      <ActionButton onClick={this.onMoveClick} glyphicon="glyphicon-pencil" type="menu" i18nKey="page-header_move" />
-    ) : (
-      ""
-    );
-    const deleteButton = page._links.delete ? (
-      <ActionButton glyphicon="glyphicon-trash" type="menu" onClick={this.onDeleteClick} i18nKey="page-header_delete" />
-    ) : (
-      ""
-    );
-    const settingsButton = page._links.edit ? (
-      <ActionButton
-        glyphicon="glyphicon-cog"
-        type="menu"
-        onClick={() => {
-          const { repository, branch } = this.props.wiki;
-          const wikiPath = `/${repository}/${branch}/`;
-          this.props.history.push(wikiPath + "settings");
-        }}
-        i18nKey="page-header_settings"
-      />
-    ) : (
-      ""
-    );
-    const restoreButton = page._links.restore ? (
-      <ActionButton
-        glyphicon="glyphicon-retweet"
-        type="menu"
-        onClick={this.onRestoreClick}
-        i18nKey="page-header_restore"
-      />
-    ) : (
-      ""
-    );
-    const createForm = (
-      <PageNameForm
-        show={this.state.showCreateForm}
-        onOk={this.onOkCreate}
-        onAbortClick={this.onAbortCreateClick}
-        labelPrefix="create"
-        directory={wiki.directory}
-      />
-    );
+    if (page && page._links) {
+      editButton = page._links.edit ? (
+        <ActionLink glyphicon="glyphicon-edit" type="menu" to="?edit=true" i18nKey="page-header_edit" />
+      ) : (
+        ""
+      );
+      historyButton = (
```

Comment:
there isn't a hal history link on the page?

Reply:
No. And Yes.
It is complicated. The original page response does not contain a HAL history link
![image](https://user-images.githubusercontent.com/6781173/148075051-2368aceb-ccb5-4d77-8e34-38b40de6dfaf.png)

Reply:
But there even more requests, which finally contains a history link. 
![image](https://user-images.githubusercontent.com/6781173/148075283-e66b7495-2514-48a0-bb1c-2ca1c4473175.png)

Reply:
But again, not in the initial response.